Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup_extension.h

```diff
@@ -77,26 +76,137 @@ duplicate_non_manifold_edges_in_polygon_soup(PointRange& points,
   return inital_nb_pts==points.size();
 }
 
+/*!
+ * \ingroup PMP_orientation_grp
+ *
+ * orients each triangle of a triangle soup using the orientation of its
+ * closest non degenerate triangle in a triangle soup.
+ *
+ * \tparam Concurrency_tag enables sequential versus parallel orientation.
+ *                         Possible values are `Sequential_tag` (the default),
+ *                         `Parallel_if_available_tag`, and `Parallel_tag`.
+ * \tparam ReferencePointRange a model of the concepts `RandomAccessContainer` whose value type is the point type.
+ * \tparam ReferenceTriangleRange a model of the concept `RandomAccessContainer`
+ *                                whose `value_type` is a model of the concept `RandomAccessContainer`
+ *                                whose `value_type` is `std::size_t`and of size 3.
+ * \tparam PointRange a model of the concepts `RandomAccessContainer`
+ *                    and `BackInsertionSequence` whose value type is the point type.
+ * \tparam TriangleRange a model of the concept `RandomAccessContainer`
+ *                       whose `value_type` is a model of the concept `RandomAccessContainer`
+ *                       whose `value_type` is `std::size_t`and of size 3.
+ *
+ * \param ref_points the points of the reference soup
+ * \param ref_faces the triangles of the reference soup
+ * \param points the points of the soup
+ * \param faces the triangles of the soup
+ * \param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamNBegin{point_map}
+ *     \cgalParamDescription{a property map associating points to the elements of the point set `points`}
+ *     \cgalParamType{a model of `ReadablePropertyMap` whose key type is the value type
+ *                    of the iterator of `PointRange` and whose value type is `geom_traits::Point_3`}
+ *     \cgalParamDefault{`CGAL::Identity_property_map<geom_traits::Point_3>`}
+ *   \cgalParamNEnd
+ *
+ *   \cgalParamNBegin{geom_traits}
+ *     \cgalParamDescription{an instance of a geometric traits class}
+ *     \cgalParamType{The traits class must provide the nested functor `Collinear_3` to check whether three points are collinear. }
```

Comment:
I guess you are missing something to compare the orientations.

Reply:
Yes, you need a lot of vector manipulation, cross product, scalar product, indeed.

I will put `Kernel`, but the problem with slapping `Kernel` the `geom_traits` is that it spreads. For example in a new package, I have one call to one `PMP` function, and making a traits concept becomes irrelevant, because if you follow what `PMP` says, you already need `Kernel` (or you have to check the `PMP` function to find what it actually requires).